storm-core/src/jvm/org/apache/storm/daemon/supervisor/BasicContainer.java

```diff
@@ -0,0 +1,494 @@
+        LOG.debug("Topology Classpath Prefix: {}", firstClasspathList);
+
+        if (object instanceof List) {
+            topoClasspath.addAll((List<String>) object);
+        } else if (object instanceof String) {
+            topoClasspath.add((String) object);
+        }
+        LOG.debug("Topology specific classpath is {}", object);
+
+        String classPath = Utils.addToClasspath(firstClasspathList, Arrays.asList(Utils.workerClasspath()));
+        String classAddPath = Utils.addToClasspath(classPath, Arrays.asList(stormJar));
+        return Utils.addToClasspath(classAddPath, topoClasspath);
+    }
+    
+    private static String substituteChildOptsInternal(String string,  String workerId, String stormId, int port, int memOnheap) {
+        if (StringUtils.isNotBlank(string)){
+            string = string.replace("%ID%", String.valueOf(port));
+            string = string.replace("%WORKER-ID%", workerId);
+            string = string.replace("%TOPOLOGY-ID%", stormId);
+            string = string.replace("%WORKER-PORT%", String.valueOf(port));
+            string = string.replace("%HEAP-MEM%", String.valueOf(memOnheap));
+        }
+        return string;
+    }
+    
+    protected List<String> substituteChildopts(Object value, String workerId, String stormId, int port, int memOnheap) {
+        List<String> rets = new ArrayList<>();
+        if (value instanceof String) {
+            String string = substituteChildOptsInternal((String) value,  workerId, stormId, port, memOnheap);
+            if (StringUtils.isNotBlank(string)){
+                String[] strings = string.split("\\s+");
+                rets.addAll(Arrays.asList(strings));
+            }
+        } else if (value instanceof List) {
+            @SuppressWarnings("unchecked")
+            List<Object> objects = (List<Object>) value;
+            for (Object object : objects) {
+                String str = substituteChildOptsInternal((String) object,  workerId, stormId, port, memOnheap);
+                if (StringUtils.isNotBlank(str)){
+                    rets.add(str);
+                }
+            }
+        }
+        return rets;
+    }
+
+    /**
+     * Launch the worker process (non-blocking)
+     * @param command the command to run
+     * @param env the environment to run the command
+     * @param processExitcallback a callback for when the process exits
+     * @param logPrefix the prefix to include in the logs
+     * @param targetDir the working directory to run the command in
+     * @return true if it ran successfully, else false
+     * @throws IOException on any error
+     */
+    protected void launchWorkerProcess(List<String> command, Map<String, String> env, 
+            String logPrefix, Utils.ExitCodeCallable<Void> processExitCallback, File targetDir) throws IOException {
+        Utils.launchProcess(command, env, logPrefix, processExitCallback, targetDir);
+    }
+    
+    @Override
+    public void launch() throws IOException {
```

Comment:
It will be good add docs around the launch method . In current 1.x branch this code takes sometime to understand